fluent-syntax/src/parser.js

```diff
@@ -179,32 +190,20 @@ export default class FluentParser {
   }
 
   getMessage(ps, comment) {
-    const id = this.getEntryIdentifier(ps);
+    const id = this.getIdentifier(ps);
 
     ps.skipInlineWS();
+    ps.expectChar("=");
 
-    let pattern;
-    let attrs;
-
-    if (ps.expectChar("=")) {
-      if (ps.isPeekPatternStart()) {
-        ps.skipIndent();
-        pattern = this.getPattern(ps);
-      } else {
-        ps.skipInlineWS();
-      }
-    }
-
-    if (id.name.startsWith("-") && pattern === undefined) {
-      throw new ParseError("E0006", id.name);
+    if (ps.isPeekPatternStart()) {
+      ps.skipIndent();
+      var pattern = this.getPattern(ps);
```

Comment:
not a huge fan of hoisting via `var` over explicit local definitions via `let`, so I'd prefer the latter.

Reply:
I'd like to keep `var`, it's what it does best.

Reply:
To elaborate: `var` is well suited for use inside of `if…else` or `try…catch` blocks and keeps to code more concise. As you know, I usually advocate for using `let` and `const` but here I feel like it's the perfect tool for the job.

I'm going AFK. If you agree and if this PR blocks you, feel free to merge it as-is. Thanks!

Reply:
I don't, but that's because I value code readability above conciseness in the `explicit over implicit` manner.

Not blocking.